### PR TITLE
[@types/dc] Fix how exceptions are exported, and add missing method

### PR DESCRIPTION
**Pull request by egrimstad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: For exceptions: https://github.com/dc-js/dc.js/commit/6283f7e9ca9d074853f7b298dfdb51254d810da4, and for `replaceFilter`: http://dc-js.github.io/dc.js/docs/html/BaseMixin.html#replaceFilter__anchor
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## dc ([<kbd>unpkg</kbd>](https://unpkg.com/browse/dc@latest/))
was missing the following properties:
1. BarChart
1. BaseMixin
1. BoxPlot
1. BubbleChart
1. BubbleMixin

<details>
<summary>as well as these 32 other properties...</summary>
<p>BubbleOverlay, CapMixin, CboxMenu, ColorMixin, CompositeChart, Config, CoordinateGridMixin, DataCount, DataGrid, DataTable, GeoChoroplethChart, HeatMap, HtmlLegend, Legend, LineChart, Logger, MarginMixin, NumberDisplay, PieChart, RowChart, ScatterPlot, SelectMenu, SeriesChart, StackMixin, SunburstChart, TextFilterWidget, baseMixin, bubbleOverlay, constants, d3Box, d3compat, renderlet</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c9fd0ce9820c4884b45ea9dc291726e05af306da
</p>

**Review by hansrwindhoff — APPROVED** [review]

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Since this is a breaking change that seems like it reflects changes first introduced into major version 4 of dc, can we take this opportunity to fix the header comment format and add a version? `@types/dc` is on version `0.0.*` while `dc` is on `4.2.*`. The discrepancy is due to the non-standard format of line 1 of `index.d.ts`.

File: types/dc/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for DCJS
+// Type definitions for dc.js 4.2
```
Comment (andrewbranch) [inline types/dc/index.d.ts:1]:
```suggestion
// Type definitions for dc 4.2
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by egrimstad** [comment]
Ready to merge

**Pull request merged**